src/Examine/LuceneEngine/Directories/ExamineDirectory.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Lucene.Net.Index;
+using static Lucene.Net.Index.IndexWriter;
+
+namespace Examine.LuceneEngine.Directories
+{
+    public abstract class ExamineDirectory : Lucene.Net.Store.Directory
+    {
+        private Action<ExamineIndexWriter> _onCommitAction;
+        public virtual void InvokeOnCommit(ExamineIndexWriter writer)
+        {
+            _onCommitAction?.Invoke(writer);
+        }
+
+        /// <summary>
+        /// Whether to write the lucene log to the local directory
+        /// </summary>
+        public bool WriteLogToLocalDirectory { get; set; }
+
+        /// <summary>
+        /// Called on commit
+        /// </summary>
+        /// <param name="action"></param>
+        public void SetOnCommitAction(Action<ExamineIndexWriter> action)
+        {
+            _onCommitAction = action;
+        }
+
+        private Func<IndexWriter, MergePolicy> _mergePolicy;
+
+        public virtual bool IsReadOnly { get; set; }
+        public virtual MergePolicy GetMergePolicy(IndexWriter writer)
+        {
+            return _mergePolicy?.Invoke(writer);
+        }
+
+        public void SetMergePolicyAction(Func<IndexWriter, MergePolicy> policy)
+        {
+            _mergePolicy = policy;
+        }
+
+        private MergeScheduler MergeScheduler { get; set; }
+        public virtual MergeScheduler GetMergeScheduler()
+        {
+            return MergeScheduler;
+        }
+
+        public void SetMergeScheduler(MergeScheduler policy)
+        {
+            MergeScheduler = policy;
+        }
+        private IndexDeletionPolicy DeletionPolicy { get; set; }
+        public IndexDeletionPolicy GetDeletionPolicy() => DeletionPolicy;
+        public void SetDeletion(IndexDeletionPolicy policy)
+        {
+            DeletionPolicy = policy;
+        }
+        public Lucene.Net.Store.Directory CacheDirectory { get; protected set; }
```

Comment:
Dont think so introducing there cacheDirectory is good idea at all, it should be managed internally in Directory, not in ExamineDirectory, as it is just base for make sure we have access to Policies and scheduler. Cache directory should be in managment of each directory (SyncTemp, S3, Blob etc ) and shouldn't be in base for directories.

Reply:
done